syft/cataloger/python/package_cataloger.go

```diff
@@ -32,33 +38,87 @@ func (c *PackageCataloger) Name() string {
 
 // Catalog is given an object to resolve file references and content, this function returns any discovered Packages after analyzing python egg and wheel installations.
 func (c *PackageCataloger) Catalog(resolver source.Resolver) ([]pkg.Package, error) {
-	// nolint:prealloc
-	var fileMatches []source.Location
+	entries, err := c.getPythonPackageEntries(resolver)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, glob := range []string{eggMetadataGlob, wheelMetadataGlob} {
-		matches, err := resolver.FilesByGlob(glob)
+	var packages []pkg.Package
+	for _, entry := range entries {
+		p, err := c.catalogEggOrWheel(entry)
 		if err != nil {
-			return nil, fmt.Errorf("failed to find files by glob: %s", glob)
+			return nil, fmt.Errorf("unable to catalog python package=%+v: %w", entry.Metadata.Location.Path, err)
+		}
+		if p != nil {
+			packages = append(packages, *p)
 		}
-		fileMatches = append(fileMatches, matches...)
 	}
 
-	var pkgs []pkg.Package
-	for _, location := range fileMatches {
-		p, err := c.catalogEggOrWheel(resolver, location)
-		if err != nil {
-			return nil, fmt.Errorf("unable to catalog python package=%+v: %w", location.Path, err)
+	return packages, nil
+}
+
+func (c *PackageCataloger) getPythonPackageEntries(resolver source.Resolver) ([]*pythonPackageData, error) {
+	var metadataLocations []source.Location
+
+	// find all primary record paths
+	matches, err := resolver.FilesByGlob(eggMetadataGlob, wheelMetadataGlob)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find files by glob: %w", err)
+	}
+	metadataLocations = append(metadataLocations, matches...)
+
+	// for every primary record path, craft all secondary record paths and build a request object to gather all file contents for each record
+	var requester = source.NewContentRequester()
+	var entries = make([]*pythonPackageData, len(metadataLocations))
+	for i, metadataLocation := range metadataLocations {
+		// we've been given a file reference to a specific wheel METADATA file. note: this may be for a directory
+		// or for an image... for an image the METADATA file may be present within multiple layers, so it is important
+		// to reconcile the RECORD path to the same layer (or the next adjacent lower layer). The same is true with
```

Comment:
"or the next adjacent lower layer"

I'm trying to make sure I understand what this means. This is the first layer to include the path as you look at the current layer and each next lower layer?

Reply:
this is in regards to using `RelativeFileByPath` which provides the "closest" path relative to the given file (with respect to the current layer or lower layers). I can adjust the existing comment.